app/assets/javascripts/host_edit_interfaces.js

```diff
@@ -320,3 +320,19 @@ function update_fqdn() {
 
   $('#hostFQDN').text(name);
 }
+
+$(document).on('change', '.interface_mac', function (event) {
+  if (event.target.id == $('#interfaceModal').find('.interface_mac').attr('id')) {
+    var mac = $('#interfaceModal').find('.interface_mac').val();
+    $.ajax({
+      type: "GET",
+      url: "/interfaces/random_name?mac=" + mac,
```

Comment:
The url should probably not be hardcoded here... I think, it would be better to store it in a data attribute using rails magic and use the data in the js.